api/src/main/java/org/pmiops/workbench/api/CloudTaskUserController.java

```diff
@@ -49,73 +55,76 @@ public class CloudTaskUserController implements CloudTaskUserApiDelegate {
           "762320479256"); // terra_prod_aou_prod_2
   private static final Logger log = Logger.getLogger(CloudTaskUserController.class.getName());
 
-  private final UserDao userDao;
-  private final CloudResourceManagerService cloudResourceManagerService;
-  private final UserService userService;
   private final AccessModuleService accessModuleService;
+  private final CloudResourceManagerService cloudResourceManagerService;
   private final InitialCreditsBatchUpdateService freeTierBillingUpdateService;
   private final InitialCreditsService initialCreditsService;
+  private final Provider<Stopwatch> stopwatchProvider;
+  private final UserService userService;
 
   CloudTaskUserController(
-      UserDao userDao,
-      CloudResourceManagerService cloudResourceManagerService,
-      UserService userService,
       AccessModuleService accessModuleService,
+      CloudResourceManagerService cloudResourceManagerService,
       InitialCreditsBatchUpdateService freeTierBillingUpdateService,
-      InitialCreditsService initialCreditsService) {
-    this.userDao = userDao;
-    this.cloudResourceManagerService = cloudResourceManagerService;
-    this.userService = userService;
+      InitialCreditsService initialCreditsService,
+      Provider<Stopwatch> stopwatchProvider,
+      UserService userService) {
     this.accessModuleService = accessModuleService;
+    this.cloudResourceManagerService = cloudResourceManagerService;
     this.freeTierBillingUpdateService = freeTierBillingUpdateService;
     this.initialCreditsService = initialCreditsService;
+    this.stopwatchProvider = stopwatchProvider;
+    this.userService = userService;
   }
 
   @Override
   public ResponseEntity<Void> auditProjectAccessBatch(List<Long> userIds) {
-    log.info(String.format("Audit project access for %d users...", userIds.size()));
```

Comment:
the bulk of this is now `auditOneUser()`

Reply:
Much cleaner! I like it!